lom/src/plugins/plugins_common/plugins_helper.go

```diff
@@ -0,0 +1,242 @@
+	periodicDetectionPluginUtil.requestAborted = false
+	periodicDetectionPluginUtil.shutDownChannel = make(chan interface{})
+	periodicDetectionPluginUtil.requestFunc = requestFunction
+	periodicDetectionPluginUtil.shutdownFunc = shutDownFunction
+	periodicDetectionPluginUtil.PluginName = pluginName
+	lomcommon.LogInfo(fmt.Sprintf("Initialized periodicDetectionPluginUtil successfuly for (%s)",pluginName))
+	return nil
+}
+
+/* This method is promoted to the plugin Periodically invokes detection logic and send heartbeat as well. Honors shutdown when shutdown is invoked on plugin */
+func (periodicDetectionPluginUtil *PeriodicDetectionPluginUtil) Request(
+	hbchan chan PluginHeartBeat,
+	request *lomipc.ActionRequestData) *lomipc.ActionResponseData {
+	if request.Timeout > 0 {
+		return GetResponse(request, "", "", ResultCodeInvalidArgument, "Invalid Timeout value for detection plugin")
+	}
+	detectionTicker := time.NewTicker(time.Duration(periodicDetectionPluginUtil.requestFrequencyInSecs) * time.Second)
+	heartBeatTicker := time.NewTicker(time.Duration(periodicDetectionPluginUtil.heartBeatIntervalInSecs) * time.Second)
+	lomcommon.LogInfo(fmt.Sprintf("Timers initialized for plugin (%s)", periodicDetectionPluginUtil.PluginName))
+        isExecutionHealthy := false
+	for {
+		select {
+		case <-heartBeatTicker.C:
+                     /* Send heartbeat only when the request is healthy */
+		     if isExecutionHealthy {
+			pluginHeartBeat := PluginHeartBeat{PluginName: periodicDetectionPluginUtil.PluginName, EpochTime: time.Now().Unix()}
+			hbchan <- pluginHeartBeat
+                     }
+		case <-detectionTicker.C:
+                     /* Perform detection logic periodically */
+		     if !periodicDetectionPluginUtil.requestAborted {
+			response := periodicDetectionPluginUtil.requestFunc(request, &isExecutionHealthy)
+			    if response != nil {
+				return response
+			    }
+		     }
+
+		case <-periodicDetectionPluginUtil.shutDownChannel:
+                        /* Shutdown stops the periodic detection */
+			lomcommon.LogInfo(fmt.Sprintf("Aborting Request for (%s)", periodicDetectionPluginUtil.PluginName))
+			responseData := GetResponse(request, "", "", ResultCodeAborted, ResultStringFailure)
+			periodicDetectionPluginUtil.requestAborted = true
+			return responseData
+		}
+	}
+}
+
+/* Shutdown that aborts the request. It also cleans up plugin defined cleanUp at the end */
+func (periodicDetectionPluginUtil *PeriodicDetectionPluginUtil) Shutdown() error {
+	lomcommon.LogInfo(fmt.Sprintf("Shutdown called for plugin (%s)", periodicDetectionPluginUtil.PluginName))
+	close(periodicDetectionPluginUtil.shutDownChannel)
+	startTime := time.Now()
+	for {
+		if periodicDetectionPluginUtil.requestAborted {
+			break
+		}
+
+		elapsedTimeInSeconds := time.Since(startTime).Seconds()
+		if elapsedTimeInSeconds > float64(periodicDetectionPluginUtil.requestFrequencyInSecs) {
+			return errors.New(fmt.Sprintf("Request could not be aborted"))
```

Comment:
I have added  !periodicDetectionPluginUtil.requestAborted so that heartbeat wont be sent out. 
I will be making enhancing the Request method as part of TODO. Will make changes in this method as well if any changes are needed.